src/lib/pgp-key.c

```diff
@@ -415,7 +406,9 @@ pgp_decrypt_seckey(const pgp_key_t *key, FILE *passfp)
     /* ask for a passphrase */
     decrypt.passfp = passfp;
     stream = pgp_new(sizeof(*stream));
-    pgp_key_reader_set(stream, key);
+    if (!pgp_key_reader_set(stream, key)) {
+        return NULL;
```

Comment:
@dewyatt, @flowher am I wrong that this function has leaks by design — it never makes `free` for stream?